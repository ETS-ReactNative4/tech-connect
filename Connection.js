import React, { Component } from 'react'
import { View, Text, Image, StyleSheet, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'
import { connect } from 'react-redux'

export class Connection extends Component {
  
  render() {

    const { connection } = this.props

    return (
      <View style={ styles.container }>
        <Image source={ require('./profile-pic.jpeg')} style={styles.picture} />
        <View style={ styles.connectionContainer }>
          <TouchableOpacity onPress={() => this.props.viewProfile(connection.id)}>
            <Text style={styles.name}>{ connection.name }</Text>
          </TouchableOpacity>
          <Text style={ styles.position }>{ connection.job_title }</Text>
          <View style={ styles.locationContainer}>
            <Icon name='map-pin' size={18} color='#4AA9C5' style={{marginRight: 7}} />
            <Text style={ styles.location }>{ connection.city }</Text>
          </View>
        </View>
      </View>
    ) 
  }
}

export const mapStateToProps = (state) => ({
  api_key: state.user.api_key
})

export default connect(mapStateToProps)(Connection)

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    display: 'flex',
    flexDirection: 'row',
    padding: 10,
    borderRadius: 5,
    marginBottom: 10,
    shadowOffset: {  width: 2,  height: 2},
    shadowRadius: 5,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  picture: {
    height: 80,
    width: 80,
    borderRadius: 40
  },
  name: {
    fontSize: 18
  },
  position: {
    color: 'gray',
    marginTop: 1
  },
  location: {
    color: 'gray',
    fontSize: 14,
    marginTop: 2
  },
  connectionContainer: {
    marginLeft: 5
  },
  locationContainer: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: 3
  }
})