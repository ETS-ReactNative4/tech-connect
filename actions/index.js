export const loginUser = (user) => ({
  type: 'LOG_IN',
  user
})

export const logoutUser = () => ({
  type: 'LOG_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
})