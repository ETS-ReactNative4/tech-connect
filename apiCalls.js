export const getLocations = async () => {
    const url = 'https://tech-connect-be.herokuapp.com/api/v1/locations'
    const response = await fetch(url)
    const locationData = await response.json()
    return locationData.data.map(location => {
      return {label: location.attributes.city}
    })
  }

export const getEmployers = async () => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/employers'
  const response = await fetch(url)
  const employerData = await response.json()
  return employerData.data.map(location => {
    return {label: location.attributes.name}
  })
}


export const getPositions = async () => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/positions'
  const response = await fetch(url)
  const positionData = await response.json()
  return positionData.data.map(location => {
    return {label: location.attributes.job_title}
  })
}

export const getAllUsers = async (key) => {
  const url = `https://tech-connect-be.herokuapp.com/api/v1/users?api_key=${key}`
  const response = await fetch(url)
  const userData = await response.json()
  return userData.data.map(user => {
    return {...user.attributes, id: user.id}
  })
}

export const getUsersFilter = async (key, filter, name) => {
  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/find?${filter}=${name}&api_key=${key}`
  const response = await fetch(url)
  const userData = await response.json()
  return userData.data.map(user => {
    return {...user.attributes, id: user.id}
  })
}

export const getUserInfo = async (id, apiKey) => {

  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/${id}?api_key=${apiKey}`
  try {
    const response = await fetch(url)
    if (!response.ok) {
      const errorText = await response.json()
      throw Error(errorText.error)
    }
    
    const userInfo = await response.json()
    const cleanedUser = {id: userInfo.data.id, ...userInfo.data.attributes}
    return cleanedUser
  } catch(err) {
    console.log(err)
  }
}