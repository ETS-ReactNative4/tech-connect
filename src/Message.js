import React, { Component } from 'react'
import { StyleSheet, View, Image, Text, TouchableOpacity } from 'react-native'
import { connect } from 'react-redux';


export class Message extends Component {
  constructor() {
    super()
    this.state = {
      showMessage: false
    }
  }

  render() {
    const { message, user } = this.props
    const connection = (user.name === message.receiver.name) ? message.sender.name :  message.receiver.name
    const sentDate = new Date(message.created_at).toUTCString().split(' ').slice(0, 4).join(' ')
    const meetingDate = new Date(message.meeting_date).toUTCString().split(' ').slice(0, 4).join(' ')
    const time = new Date(message.meeting_date).toUTCString().split(' ')[4]
    let messageText
    let messagePhoto
    let messageEnd = (message.meeting_location !== 'N/A') ? `at ${message.meeting_location} on ${meetingDate} at ${time}` : null

    if (message.receiver.name === user.name) {
      messageText = <Text style={ styles.messageBody } >You have { message.status } a meeting with { message.sender.name } { messageEnd }</Text>
      messagePhoto = message.sender.photo
    } else {
      messageText = <Text style={ styles.messageBody } >{ message.receiver.name } has { message.status } a meeting with you { messageEnd }</Text>
      messagePhoto = message.receiver.photo
    }
    return (
      <View style={ styles.messageContainer }>
        <Text style={ styles.date }>{ sentDate }</Text>
        <TouchableOpacity style={ styles.messageInfoContainer } onPress={() => this.setState({showMessage: !this.state.showMessage})}>
          <View style={ styles.imageContainer }>
            <Image source={ {uri: messagePhoto} } style={ styles.userPic }/>
          </View>
          <View style={ styles.messageInfo }>
            <Text style={ styles.name }>{ connection }</Text>
            <Text style={ styles.message }>{ message.status }</Text>
            {
              !this.state.showMessage ? null : messageText
            }
          </View>
        </TouchableOpacity>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(Message)


const styles = StyleSheet.create({
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    padding: 20,
    borderBottomWidth: 0.5,
    borderBottomColor: 'gray',
  },
  messageInfoContainer: {
    display: 'flex',
    flexDirection: 'row',
    maxWidth: '100%',
    marginRight: 10
  },
  imageContainer: {
    shadowOffset: {  width: 2,  height: 2 },
    shadowRadius: 4,
    shadowColor: 'black',
    shadowOpacity: .2,
  },
  userPic: {
    height: 50,
    width: 50,
    borderRadius: 25
  },
  date: {
    position: 'absolute',
    right: 20,
    top: 20
  },
  name: {
    fontSize: 18
  },
  message: {
    marginTop: 5,
    color: 'gray',
  },
  messageBody: {
    marginRight: 10
  },
  messageInfo: {
    marginLeft: 10,
    marginRight: 30,
    marginTop: 5
  }
})