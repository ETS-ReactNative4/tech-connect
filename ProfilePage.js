import React, { Component } from 'react'
import { View, Text, StyleSheet, TouchableHighlight, Image, Button, ScrollView } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'
import Connection from './Connection'
import { connect } from 'react-redux'
import { getUserInfo } from './apiCalls'

export class ProfilePage extends Component {
  constructor(props) {
    super(props)
  }

  viewProfile = async (id) => {
    const user = await getUserInfo(id, this.props.user.api_key)
    this.props.navigation.navigate('ProfilePage', {user})
  }

  render() {
    const user = this.props.navigation.getParam('user') ? this.props.navigation.getParam('user') : this.props.user
    
    return (
      <ScrollView style={ styles.scrollContainer }>
        <View style={ styles.container }>
          <View style={ styles.imageContainer }>
            <Image style={ styles.profilePicture } source={ require('./profile-pic.jpeg') } />
          </View>
            <View style={styles.profileContainer}>
            <View style={ styles.about }>
              <Text style={ styles.name }>{ user.name }</Text>
              <Text style={ styles.position }>{ user.position.job_title }</Text>
              <Text style={ styles.company }>{ user.employer.name }</Text>
              <View style={ styles.locationContainer}>
                <Icon name='map-pin' size={20} color='#4AA9C5' style={{marginRight: 7}} />
                <Text style={ styles.location }>{ user.location.city }</Text>
              </View>
              <Text style={ styles.bio }>{ user.bio }</Text>
              {
                this.props.user !== user && <TouchableHighlight style={styles.connectBtn}><Button title='Connect' color='white' /></TouchableHighlight>
              }
            </View>
            <Text style={ styles.connections }>Connections</Text>
            <View style={ styles.connectionsContainer }>
              {
                user.connections && user.connections.map((connection) => {
                  return <Connection id={ connection.id } viewProfile={ this.viewProfile } />
                })
              }
            </View>
          </View>
        </View>
      </ScrollView>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(ProfilePage)

const styles = StyleSheet.create({
  scrollContainer: {
    flex: 1,
    backgroundColor: '#4AA9C5',
  },
  container: {
    flex: 1,
    display: 'flex',
    alignItems: 'flex-start',
    alignItems: 'stretch',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
    paddingTop: 20,
    backgroundColor: '#4AA9C5',
    paddingBottom: 20,
  },
  imageContainer: {
    display: 'flex',
    height: 180,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 200000,
    marginBottom: 20,
    marginTop: 20,
    shadowOffset: {  width: 2,  height: 2},
    shadowRadius: 20,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  profilePicture: {
    height: 180,
    width: 180,
    borderRadius: 90,
    zIndex: 20000,
  },
  profileContainer: {
    flex: 1,
    backgroundColor: 'white',
    marginRight: 20,
    marginLeft: 20,
    marginTop: -100,
    paddingBottom: 20,
  },
  about: {
    marginTop: 90,
    display: 'flex',
    alignItems: 'center'
  },
  name: {
    fontSize: 35,
    fontWeight: '300'
  },
  position: {
    fontSize: 23,
    fontWeight: '200'
  },
  company: {
    fontSize: 21,
    fontWeight: '300',
    color: 'gray'
  },
  locationContainer: {
    display: 'flex',
    flexDirection: 'row'
  },
  location: {
    fontSize: 19,
    fontWeight: '200',
    color: 'gray'
  },
  bio: {
    fontWeight: '300',
    marginTop: 5,
    marginLeft: 10,
    marginRight: 10
  },
  connectBtn: {
    backgroundColor: '#93548F',
    paddingRight: 10,
    paddingLeft: 10,
    borderRadius: 20,
    shadowOffset: {  width: 0,  height: 2 },
    shadowRadius: 5,
    shadowColor: 'black',
    shadowOpacity: .5,
    marginTop: 10,
  },
  connections: {
    marginLeft: 20,
    marginTop: 15,
    fontSize: 15,
  },
  connectionsContainer: {
    backgroundColor: '#4AA9C5',
    marginTop: 5,
    marginLeft: 20,
    marginRight: 20,
    padding: 10,
    borderRadius: 5
  },
  language: {
    color: 'white',
    fontSize: 15,
  }
})