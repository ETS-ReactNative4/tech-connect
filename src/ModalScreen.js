import React, { Component } from 'react'
import { StyleSheet, Text, View, TextInput, Button, TouchableHighlight, TouchableOpacity } from 'react-native';
import { Input } from 'react-native-elements'
import DatePicker from 'react-native-datepicker'
import { connect } from 'react-redux'
import { sendConnectionRequest } from '../apiCalls'
import Icon from 'react-native-vector-icons/Feather'


export class ModalScreen extends Component {
  constructor() {
    super()
    this.state = {
      response: '',
      inputs: {
        datetime_1: '', 
        datetime_2: '', 
        datetime_3: '',
        meeting_1: '',
        meeting_2: '',
        meeting_3: ''   
      }
    }
  }

  sendRequest = async () => {
    const user = this.props.navigation.getParam('user')
    const response = await sendConnectionRequest(this.props.user.api_key, user.id, this.state.inputs)
    this.setState({ response })
  }

  render() {
    const { response, inputs } = this.state
    const { datetime_1, datetime_2, datetime_3, meeting_1, meeting_2, meeting_3 } = this.state.inputs
    const user = this.props.navigation.getParam('user')
    const note = `${user.name} will receive an email with your suggestions. Once confirmed you will receive a notification email.`
    const error = response.length ? <Text style={{color: 'red'}}>{response}</Text> : null
    if (response === 'Email Sent') {
      return (
        <View style={styles.container}>
          <View style={{flexDirection: 'row', alignItems: 'center', justifyContent: 'center'}}>
            <Text style={styles.title}>{ response }</Text>
            <Icon
              name='check-circle'
              size={35}
              color='#fff'
              style={{marginLeft: 5}}
            />
          </View>
          <TouchableHighlight style={styles.btnContainer}>
            <Button
              onPress={() => this.props.navigation.goBack()}
              color={'#fff'}
              title="Done"
            />
          </TouchableHighlight>
        </View>
      )
    } else {
      return (
        <View style={styles.container}>
          <Text style={styles.title}>Request a Meeting</Text>
          <Text style={styles.note}>{ note }</Text>
          <Text style={styles.suggest}>Suggest Three Date/Times:</Text>
          <DatePicker
            style={styles.date}
            date={datetime_1}
            mode='datetime'
            format='YYYY-MM-DD HH:MM:SS'
            minDate={new Date()}
            maxDate='2020-01-01'
            confirmBtnText='Confirm'
            cancelBtnText='Cancel'
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36,
                borderWidth: 0
              }
            }}
            onDateChange={date => this.setState({inputs: {...this.state.inputs, datetime_1: date}})}
          />
          <DatePicker
            style={styles.date}
            date={datetime_2}
            mode='datetime'
            format='YYYY-MM-DD HH:MM:SS'
            minDate={new Date()}
            maxDate='2020-01-01'
            confirmBtnText='Confirm'
            cancelBtnText='Cancel'
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36,
                borderWidth: 0
              }
            }}
            onDateChange={date => this.setState({inputs: {...this.state.inputs, datetime_2: date}})}
          />
          <DatePicker
            style={styles.date}
            date={datetime_3}
            mode='datetime'
            format='YYYY-MM-DD HH:MM:SS'
            minDate={new Date()}
            maxDate='2020-01-01'
            confirmBtnText='Confirm'
            cancelBtnText='Cancel'
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36,
                borderWidth: 0
              }
            }}
            onDateChange={date => this.setState({inputs: {...this.state.inputs, datetime_3: date}})}
          />
          <Text style={styles.suggest}>Suggest Three Locations:</Text>
          <Input 
            onChangeText={text => this.setState({inputs: {...this.state.inputs, meeting_1: text}})}
            containerStyle={styles.input} 
            inputContainerStyle={styles.inputContainer} 
          />
          <Input 
            onChangeText={text => this.setState({inputs: {...this.state.inputs, meeting_2: text}})} 
            containerStyle={styles.input} 
            inputContainerStyle={styles.inputContainer} 
          />
          <Input 
            onChangeText={text => this.setState({inputs: {...this.state.inputs, meeting_3: text}})}
            containerStyle={styles.input} 
            inputContainerStyle={styles.inputContainer} 
          />
          { error }
          <TouchableHighlight style={styles.btnContainer}>
            <Button 
              onPress={this.sendRequest}
              title="Send Request"
              color={'#fff'}
            />
          </TouchableHighlight>
          <TouchableHighlight style={styles.btnContainer}>
            <Button
              onPress={() => this.props.navigation.goBack()}
              color={'#fff'}
              title="Cancel"
            />
          </TouchableHighlight>
        </View>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(ModalScreen)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#4AA9C5',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 15
  },
  title: {
    fontSize: 42,
    fontWeight: 'bold',
    color: '#fff',
  },
  note: {
    marginBottom: 25,
    fontSize: 14,
    color: '#fff',
    textAlign: 'center'
  },
  suggest: {
    fontWeight: 'bold',
    fontSize: 22,
    marginTop: 20,
    color: '#fff',
    alignSelf: 'flex-start'
  },
  date: {
    width: '95%', 
    backgroundColor: '#fff', 
    borderRadius: 5, 
    marginTop: 5, 
    marginBottom: 5, 
    borderBottomWidth: 0
  },
  input: {
    width: '95%',
    borderRadius: 5, 
    backgroundColor: '#fff', 
    marginTop: 5,
    marginBottom: 5,
  },
  inputContainer: {
    borderBottomWidth: 0
  }, 
  btnContainer: {
    backgroundColor: '#93548F',
    paddingRight: 10,
    paddingLeft: 10,
    borderRadius: 20,
    shadowOffset: {  width: 0,  height: 2 },
    shadowRadius: 5,
    shadowColor: 'black',
    shadowOpacity: .5,
    marginTop: 10,
    width: 200
  }
})

