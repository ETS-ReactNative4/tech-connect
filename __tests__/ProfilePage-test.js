import React from 'react'
import { shallow } from 'enzyme'
import { ProfilePage, mapStateToProps, mapDispatchToProps } from '../src/ProfilePage'
import { TouchableHighlight, View, Button } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'
import Connection from '../src/Connection'
import { getUserInfo } from '../apiCalls'
import { logoutUser } from '../actions'


jest.mock('../apiCalls')
jest.useFakeTimers()

describe('ProfilePage', () => {
  let wrapper
  let mockUser

  beforeEach(() => {
    mockUser = {
      name: 'Howard',
      id: 10,
      position: {
        job_title: 'Developer'
      },
      employer: {
        name: 'Turing'
      },
      location: {
        city: 'Denver'
      },
      connections: [],
      phone_number: '303-333-3333',
      email: 'thisemail@email.com'
    }
    const mockNavigation = {
      navigate: jest.fn(),
      getParam: () => mockUser,
    }
    wrapper = shallow(<ProfilePage user={ mockUser } navigation={ mockNavigation } logoutUser={ jest.fn() }/>)
  })
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render a TouchableHighlight', () => {
    expect(wrapper.find(TouchableHighlight).length).toEqual(0)
  })

  it('should render a TouchableHighlight', () => {
    const mockProfile = {
      name: 'Lenard',
      position: {
        job_title: 'Developer'
      },
      employer: {
        name: 'Turing'
      },
      location: {
        city: 'Denver'
      },
      connections: []
    }
    const mockNavigation = {
      navigate: jest.fn(),
      getParam: () => mockProfile
    }
    const wrapper = shallow(<ProfilePage user={ mockUser } navigation={ mockNavigation }/>)

    expect(wrapper.find(TouchableHighlight).length).toEqual(1)
  })

  it('should render 0 connections', () => {
    expect(wrapper.find(Connection).length).toEqual(0)
  })

  it('should render 1 connection', () => {
    const mockUser = {
      name: 'Howard',
      position: {
        job_title: 'Developer'
      },
      employer: {
        name: 'Turing'
      },
      location: {
        city: 'Denver'
      },
      id: 3,
      connections: [{name: 'Howard', id: 3}]
    }
    const mockNavigation = {
      getParam: () => mockUser
    }
    const wrapper = shallow(<ProfilePage user={ mockUser } navigation={ mockNavigation } />)

    expect(wrapper.find(Connection).length).toEqual(1)
  })

  it('should have an edit icon', () => {
    const mockNavigation = {
      navigate: jest.fn(),
      getParam: () => false
    }
    const wrapper = shallow(<ProfilePage user={ mockUser } navigation={ mockNavigation }/>)

    expect(wrapper.find(Icon).length).toEqual(4)
  })

  it('should navigate to the modal screen', () => {
    const otherUser = {
      name: 'Lenard',
      position: {
        job_title: 'Developer'
      },
      employer: {
        name: 'Turing'
      },
      location: {
        city: 'Chicago'
      },
      connections: [],
      phone_number: '303-333-3333',
      email: 'thisemail@email.com'
    }
    const mockNavigation = {
      navigate: jest.fn(),
      getParam: () => otherUser,
    }
    wrapper = shallow(<ProfilePage user={ mockUser } navigation={ mockNavigation }/>)

    wrapper.find(Button).simulate('press')
    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled()
  })

  it('should call displaySettingsMenu if you are the user', () => {
    const spy = jest.spyOn(wrapper.instance(), 'displaySettingsMenu')
    wrapper.instance().forceUpdate()
    expect(spy).toHaveBeenCalled()
  })


  it('should not call displaySettingsMenu if you are not the user', () => {
    const mockOtherUser = {
      name: 'Howard',
      id: 45,
      position: {
        job_title: 'Developer'
      },
      employer: {
        name: 'Turing'
      },
      location: {
        city: 'Denver'
      },
      connections: [],
      phone_number: '303-333-3333',
      email: 'thisemail@email.com'
    }
    const mockNavigation = {
      navigate: jest.fn(),
      getParam: () => mockOtherUser,
    }
    wrapper = shallow(<ProfilePage user={ mockUser } navigation={ mockNavigation }/>)
    
    const spy = jest.spyOn(wrapper.instance(), 'displaySettingsMenu')
    wrapper.instance().forceUpdate()
    expect(spy).not.toHaveBeenCalled()
  })

  describe('viewProfile', () => {
    it('should fire getUserInfo and navigation', async () => {
      const wrapper = shallow(<ProfilePage user={ mockUser } navigation={{navigate: jest.fn(), getParam: jest.fn()}} />)
      
      await wrapper.instance().viewProfile(3)

      expect(getUserInfo).toHaveBeenCalled()
      expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled()
    })
  })

  describe('contactInfo', () => {
    it('should render a contact section', () => {
      const mockUser = {
        name: 'Howard',
        position: {
          job_title: 'Developer'
        },
        id: 3,
        employer: {
          name: 'Turing'
        },
        location: {
          city: 'Denver'
        },
        phone_number: '303-333-3333',
        email: 'thisemail@email.com',
        connections: [{name: 'Howard', id: 3}]
      }
      const wrapper = shallow(<ProfilePage user={ mockUser } navigation={{navigate: jest.fn(), getParam: jest.fn()}} />)
      wrapper.instance().contactInfo(mockUser)
  
      expect(wrapper.find(View).length).toEqual(12)
    })

    it('should not render a contact section', () => {
      const mockUser = {
        name: 'Howard',
        position: {
          job_title: 'Developer'
        },
        employer: {
          name: 'Turing'
        },
        location: {
          city: 'Denver'
        },
        phone_number: '303-333-3333',
        email: 'thisemail@email.com',
        connections: [{name: 'Howard', id: 3}]
      }
      const wrapper = shallow(<ProfilePage user={ mockUser } navigation={{navigate: jest.fn(), getParam: jest.fn()}} />)
      wrapper.instance().contactInfo(mockUser)
  
      expect(wrapper.find(View).length).toEqual(9)
    })
  })


  it('should log the user out and return to the login screen', () => {
    wrapper.instance().logOut()
    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled()
    expect(setTimeout).toHaveBeenCalled()
  })

  describe('editProfile', () => {
    it('should navigate to the EditProfile page', () => {
      const mockUser = {
        name: 'Howard',
        position: {
          job_title: 'Developer'
        },
        employer: {
          name: 'Turing'
        },
        location: {
          city: 'Denver'
        },
        phone_number: '303-333-3333',
        email: 'thisemail@email.com',
        connections: [{name: 'Howard', id: 3}]
      }
      const wrapper = shallow(<ProfilePage user={ mockUser } navigation={{navigate: jest.fn(), getParam: jest.fn()}} />)
      wrapper.instance().editProfile(mockUser)

      expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled()
    })
  })

  describe('displaySettingsMenu', () => {
    let mockUser
    let wrapper

    beforeEach(() => {
      mockUser = {
        name: 'Howard',
        position: {
          job_title: 'Developer'
        },
        id: 3,
        employer: {
          name: 'Turing'
        },
        location: {
          city: 'Denver'
        },
        phone_number: '303-333-3333',
        email: 'thisemail@email.com',
        connections: [{name: 'Howard', id: 3}]
      }
      wrapper = shallow(<ProfilePage user={ mockUser } navigation={{navigate: jest.fn(), getParam: jest.fn()}} />)
    })
    
    it('should render an edit icon', () => {
      expect(wrapper.find(Icon).length).toEqual(4)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with a user key', () => {
      const mockState = {
        user: {},
        isLoading: false,
        hasErrored: false
      }
      const expected = {
        user: {}
      }

      const result = mapStateToProps(mockState)

      expect(result).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params for logoutUser', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = logoutUser()
      const result = mapDispatchToProps(mockDispatch)

      result.logoutUser()

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})