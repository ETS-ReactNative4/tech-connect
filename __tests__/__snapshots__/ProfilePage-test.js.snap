// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage should match the snapshot 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#4AA9C5",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#4AA9C5",
        "flex": 1,
        "flexDirection": "column",
        "height": "100%",
        "paddingBottom": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "height": 180,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 20,
          "width": "100%",
          "zIndex": 200,
        }
      }
    >
      <Component
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 90,
            "height": 180,
            "width": 180,
            "zIndex": 200,
          }
        }
      />
      <ForwardRef(enhanceContext-Context.Consumer(Menu))
        onBackdropPress={[Function]}
        onClose={[Function]}
        onOpen={[Function]}
        onSelect={[Function]}
        renderer={[Function]}
        rendererProps={Object {}}
        style={
          Object {
            "position": "absolute",
            "right": 30,
            "top": 110,
            "zIndex": 100000000005,
          }
        }
      >
        <ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))
          customStyles={Object {}}
          disabled={false}
        >
          <Icon
            allowFontScaling={false}
            color="#4AA9C5"
            name="settings"
            size={24}
          />
        </ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))>
        <ForwardRef(enhanceContext-Context.Consumer(MenuOptions))
          customStyles={
            Object {
              "optionText": Object {
                "fontSize": 16,
              },
              "optionTouchable": Object {
                "activeOpacity": 80,
                "underlayColor": "#93548F",
              },
              "optionWrapper": Object {
                "alignItems": "center",
                "borderColor": "#eaeaea",
                "borderWidth": 1,
                "display": "flex",
                "height": 40,
                "justifyContent": "center",
              },
              "optionsContainer": Object {
                "width": 100,
              },
            }
          }
        >
          <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
            customStyles={Object {}}
            disableTouchable={false}
            disabled={false}
            onSelect={[Function]}
            text="Edit Profile"
          />
          <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
            customStyles={Object {}}
            disableTouchable={false}
            disabled={false}
            onSelect={[Function]}
            text="Logout"
          />
        </ForwardRef(enhanceContext-Context.Consumer(MenuOptions))>
      </ForwardRef(enhanceContext-Context.Consumer(Menu))>
    </Component>
    <Component
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "height": "100%",
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": -100,
          "paddingBottom": 20,
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "marginTop": 90,
          }
        }
      >
        <Component
          style={
            Object {
              "fontSize": 35,
              "fontWeight": "300",
            }
          }
        >
          Howard
        </Component>
        <Component
          style={
            Object {
              "fontSize": 23,
              "fontWeight": "200",
            }
          }
        >
          Developer
        </Component>
        <Component
          style={
            Object {
              "color": "gray",
              "fontSize": 21,
              "fontWeight": "300",
            }
          }
        >
          Turing
        </Component>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#4AA9C5"
            name="map-pin"
            size={20}
            style={
              Object {
                "marginRight": 7,
              }
            }
          />
          <Component
            style={
              Object {
                "color": "gray",
                "fontSize": 19,
                "fontWeight": "200",
              }
            }
          >
            Denver
          </Component>
        </Component>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#4AA9C5"
            name="linkedin"
            size={20}
            style={
              Object {
                "marginRight": 5,
              }
            }
          />
          <Component
            style={
              Object {
                "marginTop": 6,
              }
            }
          />
        </Component>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#4AA9C5"
            name="github"
            size={20}
            style={
              Object {
                "marginRight": 5,
              }
            }
          />
          <Component
            style={
              Object {
                "marginTop": 6,
              }
            }
          />
        </Component>
        <Component
          style={
            Object {
              "fontWeight": "300",
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
              "textAlign": "center",
            }
          }
        />
      </Component>
      <Component>
        <Component
          style={
            Object {
              "fontSize": 15,
              "marginLeft": 20,
              "marginTop": 15,
            }
          }
        >
          Connections
        </Component>
        <Component
          style={
            Object {
              "backgroundColor": "#4AA9C5",
              "borderRadius": 5,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
              }
            }
          >
            You have no connections.
          </Component>
        </Component>
      </Component>
    </Component>
  </Component>
</ScrollView>
`;
